File: backend/app/schemas/mess_off.py
import uuid
from datetime import date, datetime

from pydantic import BaseModel, ConfigDict, Field

from app.models.mess_off import MessOffMealType


class MessOffCreate(BaseModel):
    dates: list[date] = Field(min_length=1)
    meal_type: MessOffMealType
    reason: str | None = Field(default=None, max_length=255)


class MessOffResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    date: date
    meal_type: MessOffMealType
    reason: str | None
    created_by: uuid.UUID
    created_at: datetime

File: frontend/lib/core/models/mess_off.dart
class MessOff {
  final String id;
  final DateTime date;
  final String mealType;
  final String? reason;
  final String createdBy;
  final DateTime createdAt;

  const MessOff({
    required this.id,
    required this.date,
    required this.mealType,
    this.reason,
    required this.createdBy,
    required this.createdAt,
  });

  factory MessOff.fromJson(Map<String, dynamic> json) => MessOff(
        id: json['id'] as String,
        date: DateTime.parse(json['date'] as String),
        mealType: json['meal_type'] as String,
        reason: json['reason'] as String?,
        createdBy: json['created_by'] as String,
        createdAt: DateTime.parse(json['created_at'] as String),
      );
}
